src/nbl/video/ILogicalDevice.cpp

```diff
@@ -33,18 +33,43 @@ bool ILogicalDevice::updateDescriptorSets(uint32_t descriptorWriteCount, const I
 {
     for (auto i = 0; i < descriptorWriteCount; ++i)
     {
-        auto* ds = static_cast<IGPUDescriptorSet*>(pDescriptorWrites[i].dstSet);
-        if (!ds->processWrite(pDescriptorWrites[i]))
+        const auto& write = pDescriptorWrites[i];
+        auto* ds = static_cast<IGPUDescriptorSet*>(write.dstSet);
+
+        if (!ds->getLayout()->isCompatibleDevicewise(ds))
+            return false;
+
+        if (!ds->validateWrite(write))
             return false;
     }
 
     for (auto i = 0; i < descriptorCopyCount; ++i)
     {
-        auto* dstDS = static_cast<IGPUDescriptorSet*>(pDescriptorCopies[i].dstSet);
-        if (!dstDS->processCopy(pDescriptorCopies[i]))
+        const auto& copy = pDescriptorCopies[i];
+        const auto* srcDS = static_cast<const IGPUDescriptorSet*>(copy.srcSet);
+        const auto* dstDS = static_cast<IGPUDescriptorSet*>(copy.dstSet);
+
+        if ((!srcDS->getLayout()->isCompatibleDevicewise(srcDS)) || (!dstDS->getLayout()->isCompatibleDevicewise(dstDS)))
```

Comment:
you only want a cross-DS check, you want to check that `dstDs->isCompatibleDevicewise(srcDS)`

what you're checking right now is "true by construction"

Reply:
Yeah got it, this is also grouped in with other dumb validations I did.

Reply:
Resolved.